sound/soc/sof/pm.c

```diff
@@ -411,3 +411,21 @@ int snd_sof_prepare(struct device *dev)
 	return 0;
 }
 EXPORT_SYMBOL(snd_sof_prepare);
+
+void snd_sof_pm_runtime_enable(struct snd_sof_dev *sdev)
+{
+	/* allow runtime_pm */
+	pm_runtime_set_autosuspend_delay(sdev->dev, SND_SOF_SUSPEND_DELAY);
+	pm_runtime_use_autosuspend(sdev->dev);
+
+	/*
+	 * runtime pm for pci device is "forbidden" by default.
+	 * so call pm_runtime_allow() to enable it.
+	 */
+	pm_runtime_allow(sdev->dev);
```

Comment:
@lgirdwood the problem is here I think. I think this should be pm_runtime_allow(&sdev->pci->dev)

Reply:
but if this is a common routine it needs to deal with both the PCI and non-PCI cases, can't hard-code as suggested.

Reply:
@plbossart agree. I only tested for the pci case. We will need to rework this to address all cases